oximeter/impl/src/schema/mod.rs

```diff
@@ -153,18 +159,58 @@ fn validate_timeseries_name(s: &str) -> Result<&str, MetricsError> {
     }
 }
 
+/// Text descriptions for the target and metric of a timeseries.
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, Serialize)]
+pub struct TimeseriesDescription {
+    pub target: String,
+    pub metric: String,
+}
+
+/// Measurement units for timeseries samples.
+#[derive(Clone, Copy, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+#[serde(rename_all = "snake_case")]
+// TODO-completeness: Include more units, such as power / temperature.
+// TODO-completeness: Decide whether and how to handle dimensional analysis
+// during queries, if needed.
+pub enum Units {
+    Count,
+    Bytes,
+}
+
+impl Units {
+    /// Construct a `Units::Count`.
+    pub const fn count() -> Self {
+        Units::Count
+    }
+}
+
 /// The schema for a timeseries.
 ///
 /// This includes the name of the timeseries, as well as the datum type of its metric and the
 /// schema for each field.
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
 pub struct TimeseriesSchema {
     pub timeseries_name: TimeseriesName,
+    // TODO-cleanup: This default should be removed once schema are tracked in
+    // CockroachDB. Same for the version and scope fields.
+    #[serde(default)]
+    pub description: TimeseriesDescription,
     pub field_schema: BTreeSet<FieldSchema>,
     pub datum_type: DatumType,
+    #[serde(default = "default_schema_version")]
+    pub version: NonZeroU8,
+    #[serde(default = "AuthzScope::fleet")]
+    pub authz_scope: AuthzScope,
+    #[serde(default = "Units::count")]
+    pub units: Units,
```

Comment:
Why do these fields need `#[serde(default = ..)]` tags? The defaults don't seem super intuitive to me, but maybe that's just my unfamiliarity.

Reply:
I had previously convinced myself that we need defaults for these, to support implementing this whole thing in pieces. I don't think that's right anymore.

Mostly as a note to myself, I've written out why this is the case. The default is useful when the client expects different data from what the server produces. For example, the client expects a `units` field, but the serer doesn't provide one. We're not in that situation. Here, the server produces new fields that the client isn't aware of. Older clients can ignore these new fields. And newer clients which expect this field can only exist once we've deployed the code that produces it. Defaults aren't needed in either case.

Part of my confusion is around the databases. Currently `oximeter` derives a schema for each sample, and inserts it in ClickHouse. But we don't have columns for things like the `units` yet. However, we still don't need a `#[serde(default)]`, because we use the internal `DbTimeseriesSchema` type to model the database field. In the conversion from that to the real schema type, we can insert the default, but that's not a `#[serde(default)]`.

Anyway, sorry for the confusion on this, and thanks for bringing this up again. I'll remove all the default attributes and implementations that I've added here.

Reply:
I removed the `#[serde(default)]` attributes and the constructors used to create the defaults in 5610bb467

Reply:
Hmm, so we _do_ actually need the defaults for [this test](https://buildomat.eng.oxide.computer/wg/0/details/01J0PT87YHXSPPPTX7ZXQ3NZCT/m6KJAqWpc4klzlu2Akw78hwN89jbzUThRPeG33pW6mN4Htb5/01J0PT8KF46B6ERA5ZZRBECTQS#S5148). That test will be obviated by a lot of this work here, so I might remove it now. But I will dig into it a bit first.

Reply:
I fixed this by just overwriting the test file to include the new fields, in 13048c1333cd8b4f828a45522fd0f886879be6ec